librz/core/cmd/cmd_eval.c

```diff
@@ -271,226 +237,181 @@ RZ_IPI void rz_core_theme_nextpal(RzCore *core, int mode) {
-			RzListIter *th_iter;
-			const char *th;
-			rz_list_foreach (themes_list, th_iter, th) {
-				rz_cons_printf("%s\n", th);
-			}
-		} else {
-			RzList *themes_list = rz_core_list_themes(core);
-			RzListIter *th_iter;
-			const char *th;
-			rz_list_foreach (themes_list, th_iter, th) {
-				if (curtheme && !strcmp(curtheme, th)) {
-					rz_cons_printf("> %s\n", th);
-				} else {
-					rz_cons_printf("  %s\n", th);
-				}
-			}
-		}
-		break;
-	case 's': rz_cons_pal_show(); break; // "ecs"
-	case '*': rz_cons_pal_list(1, NULL); break; // "ec*"
-	case 'h': // echo
-		if (input[1] == 'o') {
-			rz_core_echo(core, input + 2);
-		} else {
-			rz_cons_pal_list('h', NULL);
-		}
+	switch (mode) {
+	case RZ_OUTPUT_MODE_RIZIN:
+		rz_cons_pal_list(1, NULL);
 		break;
-	case 'j': // "ecj"
+	case RZ_OUTPUT_MODE_JSON:
 		rz_cons_pal_list('j', NULL);
 		break;
-	case 'c': // "ecc"
-		rz_cons_pal_list('c', input + 1);
-		break;
-	case '\0': // "ec"
+	case RZ_OUTPUT_MODE_STANDARD:
 		rz_cons_pal_list(0, NULL);
 		break;
-	case 'r': // "ecr"
-		rz_cons_pal_random();
-		break;
-	case 'n': // "ecn"
-		rz_core_theme_nextpal(core, 'n');
-		break;
-	case 'p': // "ecp"
-		rz_core_theme_nextpal(core, 'p');
+	default:
+		return RZ_CMD_STATUS_ERROR;
+	};
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_eval_color_display_palette_css_handler(RzCore *core, int argc, const char **argv) {
+	rz_cons_pal_list('c', argv[1]);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_eval_color_set_default_palette_handler(RzCore *core, int argc, const char **argv) {
+	rz_cons_pal_init(core->cons->context);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_eval_color_set_random_palette_handler(RzCore *core, int argc, const char **argv) {
+	rz_cons_pal_random();
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_eval_color_set_colorful_palette_handler(RzCore *core, int argc, const char **argv) {
+	rz_cons_pal_show();
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_eval_color_load_previous_theme_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_theme_nextpal(core, 'p');
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_eval_color_load_next_theme_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_theme_nextpal(core, 'n');
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_eval_color_highlight_list_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	RzCmdStateOutput state = { 0 };
+	rz_cmd_state_output_init(&state, RZ_OUTPUT_MODE_STANDARD);
+	rz_core_meta_print_list_all(core, RZ_META_TYPE_HIGHLIGHT, &state);
+	rz_cmd_state_output_print(&state);
+	rz_cmd_state_output_fini(&state);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_eval_color_load_theme_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+
+	RzList *themes_list = NULL;
+	RzListIter *th_iter;
+	const char *th;
+	if (argc == 2) {
+		rz_core_load_theme(core, argv[1]);
```

Comment:
Please check the result of `rz_core_load_theme()` and return using the function `bool2status()`

Reply:
I can't seem to call `bool2status()` from `cmd_eval.c` due to the implicit declaration error since `cmd_info.c` isn't compiled before `cmd_eval.c`, I think.
How should get around that? Maybe we can declare it somewhere?
 
Meanwhile, I was wondering about such a workaround:
```suggestion
		bool res = rz_core_load_theme(core, argv[1]);
		return res ? RZ_CMD_STATUS_OK : RZ_CMD_STATUS_ERROR;
```

Reply:
@ret2libc maybe make `bool2status()` an `RZ_IPI`?

Reply:
i suggest to just re-define it in your file. Hopefully the compiler will just compile everything away as it is a very basic typing thing. Moving it to RZ_IPI would probably prevent such optimizations.